### ISSUE TITLE
Semantic Release is not updating the `master` branch version

### ISSUE THREAD

**Issue by ashblue** [ashblue/fluid-behavior-tree#32]
Releases are not updating the root `package.json` version. Investigate why. Open a ticket with Oyster or the package assembler if the issue is in the corresponding package.

**Comment by favoyang** [comment]
I guess the reason could be you're running the npm plugin against the generated `dist/` folder. The original package.json is not get updated. So you need to tweak the ordering, by updating package.json at the right place first, then run the build script. You also need to commit the changed package.json via Git plugin.

Here's a working .releaserc: 
https://github.com/favoyang/unity-addressable-importer/blob/master/.releaserc.json (json format, but simliar)

**Comment by favoyang** [comment]
Please try install semantic-release and run it with --dryrun to verify your modification https://semantic-release.gitbook.io/semantic-release/usage/configuration#dryrun

**Comment by ashblue** [comment]
Well it's working in some of my other repos so I'll have to do a little digging. Yeah, it could quite possibly be the ordering in the release file.

**Comment by ashblue** [comment]
Finally had some time to dig into this. The issue is directly tied to Oyster Package Manager. It seems a version change on the Git semver plugin requires specifying the package.json to check it in. Need to do some more digging there.

https://github.com/ashblue/oyster-package-generator

**Comment by favoyang** [comment]
You need to specify the `assets` parameter of @semantic-release/git (in .releaserc.json) if the package is not located at the root path.

https://github.com/semantic-release/git

And if the CHANGELOG is not located at the root path, you need to specify the `changelogFile` path of @semantic-release/changelog as well.

https://github.com/semantic-release/changelog

**Comment by ashblue** [comment]
I found the issue and it has to do with the NPM plugin running after the Git plugin for Semantic Release. I'm also having to rewrite the build compiler to hard write version values to `master` before deploy. Going to take me some time to get the CLI tool stable to do this via Oyster (WIP).

**Comment by favoyang** [comment]
Nice catch.

**Comment by ashblue** [comment]
This issue should now be resolved as of #59

**Comment by favoyang** [comment]
> This issue should now be resolved as of #59

@ashblue thanks for the update. So everything should be settled down once this been merged into the master branch?

**Comment by favoyang** [comment]
And am I right that you will not change the package name of these `package.json` files to make it different?
- https://github.com/ashblue/fluid-behavior-tree/blob/develop/package.json `com.fluid.behavior-tree`
- https://github.com/ashblue/fluid-behavior-tree/blob/develop/Assets/com.fluid.behavior-tree/package.json `com.fluid.behavior-tree`

**Comment by ashblue** [comment]
The underlying package manager Oyster relies on the root package.json for a lot of operations. As the package manager itself is written in Node.js (same package tech Unity opted into). Multiple package.json files aren't unusual in Node.js projects. It could be renamed or removed into a separate branch, but it would be a lot of work to do so. If there was some sort of underlying config file to specify the pacakge.json I could easily fix this. It would have to be that or a pull request on Oyster to create a separate branch at the right time in the template files. Not against the idea at all, just don't have time to write the feature atm (pandemic was not kind to me the past year).

**Comment by favoyang** [comment]
https://github.com/ashblue/fluid-behavior-tree/blob/develop/Assets/com.fluid.behavior-tree/package.json is the actually upm package meta file, so that file is all I care about. I'm just wondering after #59 gets merged into the master branch (from the develop branch), will that file get version synced (`x.y.z` instead of `0.0.0`)? If so, I may have a way to resolve the duplicate package.json issue on the OpenUPM side, by introducing a "package meta file search path".

**Comment by ashblue** [comment]
Once a new release triggers it should start auto-updating the `package.json` versions automatically in `master`. Fluid Stats is doing this already and everything seems to be working fine https://github.com/ashblue/fluid-stats/blob/master/Assets/com.fluid.stats/package.json. I've updated this repo to Oyster v2 already in `develop`. Releases won't trigger unless a new feature of bugfix is detected.

**Comment by favoyang** [comment]
Okay. I will watch for the next release to verify it.

**Comment by ashblue** [comment]
Thx for your patience on this. Took up a lot more time than expected to create a migration tool to get my old repos from Oyster v1 to v2 😅
